File: OCTOBER-2025/01-10-2025/p1518.java
class Solution {
    public int numWaterBottles(int bottle, int exchange) {
       int ans=bottle;
       int empty=bottle;
       while(empty>=exchange){
        int filled=empty/exchange;
        ans+=filled;
        empty=(empty%exchange)+filled;
       }
        return ans;
    }
}

File: OCTOBER-2025/01-10-2025/p1518.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        ne=numExchange
        nb=numBottles
        c=nb
        if nb<ne:
            return nb
        while nb>=ne:
            c+=nb//ne
            nb=nb//ne+nb%ne
        return c